### PR TITLE
Added Git checkout option for GitRepoClone

### PR DESCRIPTION
**Pull request by RakeshwarK**

### CODE REVIEW

File: src/VirtualClient/VirtualClient.Dependencies/Packaging/GitRepoClone.cs [hunk 1]
```diff
@@ -100,6 +117,25 @@ await systemManagement.PackageManager.RegisterPackageAsync(
                 await systemManagement.PackageManager.RegisterPackageAsync(package, cancellationToken)
                     .ConfigureAwait(false);
             }
+
+            if (this.Checkout != string.Empty)
+            {
+                using (IProcessProxy checkoutProcess = processManager.CreateProcess("git", $"-C {cloneDirectory} checkout {this.Checkout}", this.PlatformSpecifics.PackagesDirectory))
```
Comment (yangpanMS) [inline src/VirtualClient/VirtualClient.Dependencies/Packaging/GitRepoClone.cs:123]: Use ExecuteCommandAsync on VirtualClientCompnentExtensions unless you are doing something special

File: website/docs/dependencies/0040-install-git-repo.md [hunk 2]
```diff
@@ -36,6 +37,7 @@ In this example, VC clones https://github.com/eembc/coremark.git into the runtim
       "Type": "GitRepoClone",
       "Parameters": {
           "RepoUri": "https://github.com/eembc/coremark.git",
+          "Checkout": "eefc986"
```
Comment (yangpanMS) [inline website/docs/dependencies/0040-install-git-repo.md:40]: Thanks, as part of this I would propose you add the Checkout to workloads like coremark. Find a good release commit

**Review by yangpanMS — APPROVED** [review]

File: src/VirtualClient/VirtualClient.Dependencies/Packaging/GitRepoClone.cs [hunk 3]
```diff
@@ -47,13 +47,30 @@ public Uri RepoUri
...
+        }
+
         /// <summary>
         /// Executes the git clone operation.
         /// </summary>
         protected override async Task ExecuteAsync(EventContext telemetryContext, CancellationToken cancellationToken)
         {
             telemetryContext.AddContext("repoUri", this.RepoUri);
             telemetryContext.AddContext("packagesDirectory", this.PlatformSpecifics.PackagesDirectory);
+            telemetryContext.AddContext("checkout", this.Checkout);
```
Comment (brdeyo) [inline src/VirtualClient/VirtualClient.Dependencies/Packaging/GitRepoClone.cs:73]: Let's use the actual terms that are related to Git for this feature. You can expose both a "Commit" property as well as a "Branch" property.

This makes it more clear to a user/reader of a profile.

**Review by brdeyo — CHANGES_REQUESTED** [review]

File: src/VirtualClient/VirtualClient.Dependencies/Packaging/GitRepoClone.cs (hunk 3 above)
Comment (yangpanMS) [reply src/VirtualClient/VirtualClient.Dependencies/Packaging/GitRepoClone.cs:73]: so "Commit" but you could use branch, or commit or tag

Comment (brdeyo) [reply src/VirtualClient/VirtualClient.Dependencies/Packaging/GitRepoClone.cs:73]: Good points. Let's just stay with "Checkout" so long as we have good documentation describing the possible uses. I retract my statements above.

**Review by brdeyo — APPROVED** [review]

File: website/docs/dependencies/0040-install-git-repo.md [hunk 4]
```diff
@@ -21,6 +21,7 @@ The following section describes the parameters used by the individual component
 |---------------|--------------|-----------------------------------------------------------------------------------------------------------------|
 | PackageName   | Yes          | The logical name of the that will be registered with the Virtual Client runtime to represent the packages directory into which the repo was cloned. Other profile components can use this name to reference/discover the repo and its location. |
 | RepoUri       | Yes          | The full URI to the Git repository to download/clone into the packages directory.                               |
+| Checkout      | No           | The branch-name or commit-hash or tag to checkout to a specific branch, commit, tag in the repository.          |
```
Comment (yangpanMS) [inline website/docs/dependencies/0040-install-git-repo.md:24]: check out also supports tag

Comment (RakeshwarK) [reply website/docs/dependencies/0040-install-git-repo.md:24]: yes, it supports tags.

**Review by brdeyo — APPROVED** [review]

**Pull request merged**